sycl/doc/design/CommandGraph.md

```diff
@@ -0,0 +1,62 @@
+# Command Graph Extension
+
+This document describes the implementation design of the 
+[SYCL Graph Extension](https://github.com/intel/llvm/pull/5626).
+
+A related presentation can be found 
+[here](https://www.youtube.com/watch?v=aOTAmyr04rM).
+
+## Requirements
+
+An efficient implementation of a lazy command graph execution and its replay
+requires extensions to the PI layer. Such an extension is command buffers,
+where a command-buffer object represents a series of operations to be enqueued
+to the backend device and their dependencies. A single command graph can be
+partitioned into more than one PI command-buffer by the runtime.
+We distinguish between backends that support command buffer extensions and
+those that do not. Currently command buffer extensions are only supported by
+Level Zero. All other backends would fall back to an emulation mode.
+
+The emulation mode targets support of functionality only, without potentially
+resulting performance improvements, i.e. execution of a closed Level Zero
+command list multiple times. 
+
+#### Command Buffer extension
+
+| Function | Description |
+| ------------------------- | ------------------------ |
+| `piextCommandBufferCreate` | create a command-buffer. |
+| `piextCommandBufferRetain` | incrementing reference count of command-buffer. |
+| `piextCommandBufferRelease` | decrementing reference count of command-buffer. |
+| `piextCommandBufferFinalize` | no more commands can be appended, makes command buffer ready to enqueue on command-queue. |
+| `piextCommandBufferNDRangeKernel` | append a kernel execution command to command buffer. |
```

Comment:
We also now have:
* piextCommandBufferMemcpyUSM - append a USM memcpy command to the command-buffer
* piextCommandBufferMemBufferCopy - append a mem buffer copy command to the command-buffer
* piextCommandBufferMemBufferCopyRect - append a rectangular mem buffer copy command to the command-buffer

Reply:
done